### DISCUSSION TITLE
How to have an 'internal only' package in a workspace

### DISCUSSION THREAD

**Discussion by ieedan** [jsr-io/jsr#772] [Q&A] [answered]
For instance I have a workspace that has 5 packages.

`deno.json`:
```jsonc
{
	"tasks": {},
	"fmt": {
		"useTabs": true,
		"semiColons": true,
		"indentWidth": 4
	},
	"importMap": "import_map.json",
	"workspace": ["packages/parsing", "packages/cli", "packages/analyzers", "packages/utils", "packages/highlight"]
}
```

I want everything to be published with the exception of `packages/utils` which is just shared internal utilities for the rest of the packages.

The problem is that when I go to publish even though everything works locally I get an error because it cannot resolve `@my-scope/utils` from jsr.

I have tried added it in the workspace import map like this:
```jsonc
{
    "imports": {
        "@utils": "./packages/utils/src/index.ts"
    }
}
```

and then referencing it like this:
```ts
import { x } from "@utils"; 
```

but then when I go to publish I get this error:
```
failed to build module graph: Module not found "file:///utils/src/index.ts"
```

**Comment by ieedan** [comment]
It also does the same thing when I try and use a relative import such as:
```ts
import { x } from "../../utils/src/script.ts";
```

**Comment by crowlKats** [comment, marked as answer]
@ieedan if the package description starts with `INTERNAL`, it will not be displayed in the package search. it currently still appears in the scope packages list, but i will fix that in the coming days.

**Comment by ieedan** [comment]
This would still require me to publish the code correct? And then by description you mean the description you enter on the JSR site or `@scope/INTERNAL_package`?

**Reply by crowlKats** [reply]
yes, you would still need to publish it. and i mean the description you enter on the JSR site

**Reply by ieedan** [reply]
Ah okay that makes sense thanks!
